File: iniciante/leetcode/two_sum/two_sum.py

# Usar mapa de hash para pegar complexidade melhor (Dicas do lletcodee)

class Solution(object):
    def twoSum(self, nums, target):
        
        mapa = {}

        for index, num in enumerate(nums):
            
            complemento = target - num

            if mapa.get(complemento) != None:
                return [mapa[complemento], index]

            mapa[num] = index


#teste

solution = Solution()

print(solution.twoSum([3,3],6))

File: iniciante/leetcode/two_sum/two_sum.js

// Usar mapa de hash para pegar complexidade melhor (Dicas do lletcodee)

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    let result = []

    nums.forEach((num, index, nums) => {
        
        for(i = index + 1; i < nums.length; i++){

            if(target === (num + nums[i])){
                result = [index, i]
            }
        }

    })

    return result 

};

// Teste

nums = [3,3]
target = 6

console.log(twoSum(nums, target))